src/gs-category-page.c

```diff
@@ -342,15 +342,28 @@ gs_category_page_create_filter_list (GsCategoryPage *self,
 			 * placeholder app tiles get cleared out, then set
 			 * "empty state" message for an empty USB disk
 			 */
-			if (g_strcmp0 (gs_category_get_id (category), "usb") == 0) {
-				gs_category_page_populate_filtered (self, s);
+			if (g_strcmp0 (gs_category_get_id (category), "usb") != 0) {
+				gtk_widget_set_visible (self->no_apps_box, category_is_usb);
+				gtk_widget_set_visible (self->scrolledwindow_category, 
+							!category_is_usb);
+
+				continue;
+			} else {
+				GPtrArray *usb_apps;
 				category_is_usb = TRUE;
+				gs_category_page_populate_filtered (self, s);
+				/* After reloading the page it will get the actual apps for the
+                                 * for the usb so we test if the category will truly have the 
```

Comment:
It looks like this and the following lines aren't aligned with the `*` in the initial line